test/cloudant.test.js

```diff
@@ -230,19 +243,19 @@ describe('cloudant connector', function() {
       });
       it('returns error missing data type when sorting', function(done) {
         CustomerSimple.find({where: {'address.state': 'CA'},
-        order: 'address.city DESC'},
+        order: 'address.state DESC'},
           function(err, customers) {
             should.exist(err);
-            err.message.should.match(/Unspecified or ambiguous sort type/);
+            err.message.should.match(/no_usable_index,missing_sort_index/);
             done();
           });
       });
-      it('returns result when sorting type provided - missing first level' +
+      it('returns result when sorting type provided - missing first level ' +
```

Comment:
see comment https://github.com/strongloop/loopback-connector-cloudant/pull/163/files#r134341551
I think you might be affected by the code in this test case, while this one aimed to make sure my "field type detecting" algorithm in `buildSort` doesn't throw error if the property doesn't exist in modelDef :-p